File: address-geocode-canada-python-examples/get_geo_location_rest_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../address-geocode-canada-python/REST"))

from get_geo_location_rest import get_geo_location


def get_geo_location_rest_sdk_go(is_live: bool, license_key: str) -> None:
    print("\r\n----------------------------------------------------")
    print("Address Geocode - Canada - GetGeoLocation - REST SDK")
    print("----------------------------------------------------")

    address = "1053 Carling Ave"
    municipality = "Ottawa"
    province = "ON"
    postal_code = "K1Z 7K4"
    timeout_seconds = 15

    print("\r\n* Input *\r\n")
    print(f"Address     : {address}")
    print(f"Municipality: {municipality}")
    print(f"Province    : {province}")
    print(f"PostalCode  : {postal_code}")
    print(f"License Key : {license_key}")
    print(f"Is Live     : {is_live}")

    try:
        response = get_geo_location(address, municipality, province, postal_code, license_key, is_live)

        print("\r\n* Geocode Info *\r\n")
        if response and not response.Error:
            print(f"Latitude  : {response.Latitude}")
            print(f"Longitude : {response.Longitude}")
            print(f"PostalCode: {response.PostalCode}")
            print(f"MatchCode : {response.MatchCode}")
        else:
            print("No geocode info found.")

        if hasattr(response, "Error") and response.Error:
            print("\r\n* Error *\r\n")
            print(f"Error Desc    : {response.Error.Desc}")
            print(f"Error Number  : {response.Error.Number}")
            print(f"Error Location: {response.Error.Location}")

    except Exception as e:
        print("\r\n* Error *\r\n")
        print(f"Error Message : {str(e)}")

File: address-geocode-canada-nodejs-examples/get_geo_location_rest_sdk_example.js
import { GetGeoLocationClient } from "../address-geocode-canada-nodejs/REST/get_geo_location_rest.js";

async function GetGeoLocationRestGo(licenseKey, isLive) {
    console.log("\n----------------------------------------------------");
    console.log("Address Geocode - Canada - GetGeoLocation - REST SDK");
    console.log("----------------------------------------------------");

    const address = "1053 Carling Ave";
    const municipality = "Ottawa";
    const province = "ON";
    const postalCode = "K1Z 7K4";
    const timeoutSeconds = 15;

    console.log("\n* Input *\n");
    console.log(`Address        : ${address}`);
    console.log(`Municipality   : ${municipality}`);
    console.log(`Province       : ${province}`);
    console.log(`PostalCode     : ${postalCode}`);
    console.log(`License Key    : ${licenseKey}`);
    console.log(`Is Live        : ${isLive}`);
    console.log(`Timeout Seconds: ${timeoutSeconds}`);

    try {
        const response = await GetGeoLocationClient.invoke(
            address,
            municipality,
            province,
            postalCode,
            licenseKey,
            isLive,
            timeoutSeconds
        );

        console.log("\n* Geocode Info *\n");
        if (response) {
            console.log(`Latitude  : ${response.Latitude}`);
            console.log(`Longitude : ${response.Longitude}`);
            console.log(`PostalCode: ${response.PostalCode}`);
            console.log(`MatchCode : ${response.MatchCode}`);
        } else {
            console.log("No geocode info found.");
        }

        if (response.Error) {
            console.log("\n* Error *\n");
            console.log(`Error Desc    : ${response.Error.Desc}`);
            console.log(`Error Number  : ${response.Error.Number}`);
            console.log(`Error Location: ${response.Error.Location}`);
            return;
        }
    } catch (e) {
        console.log("\n* Error *\n");
        console.log(`Error Message: ${e.message}`);
    }
}

export { GetGeoLocationRestGo };
